src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -472,7 +495,33 @@ public MethodIL GetMethodILWithInlinedSubstitutions(MethodIL method)
                 debugInfo = new SubstitutedDebugInformation(debugInfo, sequencePoints.ToArray());
             }
 
-            return new SubstitutedMethodIL(method, newBody, newEHRegions.ToArray(), debugInfo);
+            ArrayBuilder<string> newStrings = default;
+            if (hasGetResourceStringCall)
+            {
+                for (int offset = 0; offset < flags.Length; offset++)
+                {
+                    if ((flags[offset] & OpcodeFlags.GetResourceStringCall) == 0)
+                        continue;
+
+                    Debug.Assert(newBody[offset] == (byte)ILOpcode.call);
+                    var getter = (EcmaMethod)method.GetObject(new ILReader(newBody, offset + 1).ReadILToken());
+
+                    string resourceString = GetResourceStringForAccessor(getter);
+                    if (resourceString == null)
+                        continue;
+
+                    int tokenValue = NewStringTokenRidStart - newStrings.Count;
+                    newStrings.Add(resourceString);
+
+                    newBody[offset] = (byte)ILOpcode.ldstr;
+                    newBody[offset + 1] = (byte)tokenValue;
+                    newBody[offset + 2] = (byte)(tokenValue >> 8);
+                    newBody[offset + 3] = (byte)(tokenValue >> 16);
+                    newBody[offset + 4] = 0x70;
```

Comment:
Is the call instruction the same size as the ldstr? If yes, I think it's worth adding a comment (and maybe advancing the offset so that the for loop doesn't have to go over the 4 bytes one by one. If not, then this looks wrong.

Reply:
Added comment. Yep, they're the same shape/size.